path/cache.go

```diff
@@ -0,0 +1,19 @@
+package path
+
+import "path/filepath"
+
+// CachedSignature returns the cached signature file path
+func CachedSignature(
+	manifestDigestAlgorithm,
+	manifestDigest,
+	blobDigestAlgorithm,
+	blobDigest string) string {
```

Comment:
You can accept `notation.Descriptor` for both manifest and blob to keep this function signature cleaner.